### PR TITLE
DOC: Add information about StyleFrame, to format Excel output

### LINKED ISSUE #47525: DOC: Mention "StyleFrame" package in "DataFrame.to_excel()"
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_excel.html

### Documentation problem

It is not a "problem" but enhancement.

### Suggested fix for documentation

Do suggest the [StyleFrame](https://styleframe.readthedocs.io/) package to be more flexibel with formating the excel output.

btw: I won't suggest to integrate StyleFrame into pandas. Formatting export outputs is not the job of pandas and I assume you agree with this. But is is an often mentioned "problem" in the internet community that the pandas excel output lacks of the possibility to formatting.

### PR DESCRIPTION
**Pull request by INDIG0N**
Closes #47525 
The issue mentions altering the ecosystem page on the pandas website, do you guys know how I would go about doing that?

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2189,6 +2189,9 @@ def to_excel(
         Once a workbook has been saved it is not possible to write further
         data without rewriting the whole workbook.
 
+        If you'd like to be more flexible with the output formatting,
```
Comment (mroeschke) [inline pandas/core/generic.py:2192]: As mentioned in https://github.com/pandas-dev/pandas/issues/47525#issuecomment-1169199316=, you can update the ecosystem docs in `doc/source/ecosystem.rst` to mention StyleFrame and then use a Sphix reference to mention there are ecosystem packages to help with output formatting

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (INDIG0N) [reply pandas/core/generic.py]: Cool, tahnks. I added the package to the ecosystem doc and used a Sphinx reference in the docstring.

Comment (mroeschke) [reply pandas/core/generic.py]: Looks like `StyleFrame` needs to be referenced in that doc somewhere

**Comment by datapythonista** [comment]
Thanks @INDIG0N. The idea would be to add a section for `StyleFrame` in our Ecosystem page in the web.

I personally don't see how we can add a useful note to the `to_excel` docstring if we don't link to the project (which seems reasonable). I don't think it makes a lot of sense adding to the docstring "Check the ecosystem page for packages that can be useful" or "There is a package to add formatting to the exported Excel files, but you need to check the Ecosystem page". If we don't want to link third-party packages we don't know much about in our docs other than the Ecosystem, I think let's just add the entry to the Ecosystem page.

You can see other entries in Ecosystem to see what's expected to be added.

**Comment by attack68** [comment]
The `StyleFrame` documentation does not make it clear to me the specific features, in particular the specific features beyond `pandas.Styler` own capabilities, which for example can already style:

- background colors,
- text colors,
- font styles,
- borders,
- number formats,

I think if we add a third party library to ecosystem, it would be worthwhile stating what it can do, extra, that pandas cannot.

In particular that warrants the use of the phrase "easier styling", which, currently, I am skeptical of.

**Comment by datapythonista** [comment]
I'm not aware of pandas providing much functionality for exporting an excel file with colors, number formats... `StyleFrame` is a replacement for `to_excel` that does provide a way to set all the formats in a wrapper of a `DataFrame` that then can be exported to an Excel file.

I agree that just saying "easier styling" can be improved, but I think it's worth having it in our ecosystem.

@INDIG0N do you mind expanding the description a bit to make it clear for readers what `StyleFrame` is about? I think this PR is ready just having a bit clearer description. Also, if you want to write a very short example, that could also help. You have one here: https://github.com/DeepSpace2/StyleFrame#simple-example I think we should have a much simpler one, maybe with just one column and changing the background color or something. But if it's complex, just with a bit clearer description is also fine.

And sorry it's taking quite long to review this PR.

**Comment by attack68** [comment]
Exporting from Styler to Excel is documented in the Table Visualization user guide. It includes coloring, borders and fonts for headers, index and data cells. It can also write the excel number format directly. The conditional formatting tools are all directly tied into Styler.

Im not against including StyleFrame but I reiterate that if it can only replicate what Styler can already do there is no point in its inclusion. I would like to kmow more about its additional features.

**Comment by datapythonista** [comment]
Ah, thanks for the clarification @attack68, I didn't know our Styler supported exporting to Excel, my bad. I agree with you, we should say in the description why it can make sense to use StyleFrame instead of our styler then. @INDIG0N do you know whay StyleFrame supports that we don't? Or can you research about it?

**Comment by INDIG0N** [comment]
Sorry it took me so long to get back to this @datapythonista

I've been having trouble finding anythin styleframe can do that the pandas styler can;t, so at this point I guess we should just close the pull request and the issue. Do you guys know how I can close this PR or is that something that has to be done your end?

**Comment by attack68** [comment]
close as requested

**Pull request closed without merging**